odoo/addons/test_lint/tests/_odoo_checker_sql_injection.py

```diff
@@ -45,6 +45,8 @@ def _allowable(self, node):
         return (isinstance(node, astroid.Attribute)
             and isinstance(node.expr, astroid.Name)
             and node.attrname.startswith('_')
+            # cr.execute('SELECT * FROM %s' % 'table') is OK since that is a constant
+            or isinstance(node, astroid.Const)
```

Comment:
Seems unnecessary and in fact specifically something we don't want to allow, the developer should just fold the constant into the query directly.

The one justification here would be for a top-level const, but that would require deref'ing a `Name` to see what's on the other side.

And if that were to be done it should be at the top of the method, before checking for `Call`.

Reply:
Sorry, I didn't get you

I'm detecting the following cases as valid (no-sql-injection):
  - https://github.com/odoo/enterprise/blob/2d5e2fbca7e7275929ef7c9ed6074c419e3e1658/l10n_de_reports/models/datev_export_csv.py#L135-L138

It means, if the args or kwargs are constants so it is not a sql-injection since that a constant can't be injected

Examples:

```python
self.env.cr.execute("SELECT * FROM %s" % 'table_constant')
self.env.cr.execute("SELECT * FROM {}".format('table_constant'))
self.env.cr.execute("SELECT * FROM %(table_constant)s" % {'table_variable': 'table_constant'})
```

Without this the sql-injection will raises this check for this code

I am even following your comment here:
 - https://github.com/odoo/odoo/pull/76185#discussion_r705929177
   - `After looking at the CI failure, constants should probably be allowed as parameters.`

So, I don't get this new comment, sorry

Reply:
Nah it's fine sorry about that, I just got confused because I got focused on the (somewhat simplistic) example.